### PR TITLE
KeycloakContainer and Keycloak service refactoring, Keycloak image bump and resulting OCP fixes

### PR DESCRIPTION
**Pull request by michalvavrik**
### Summary

- I'm updating Keycloak image in TS and there is repeated pattern as I can't use system property in image name. This should help. 
- Running tests for `quay.io/keycloak/keycloak:14.0.0` seems beyond the point now as it is not supported in community or product. It makes sense to keep branches like 3.2 in TS to verify support for legacy images, but here we test upstream and I don't think KC 14 support is expected.
- Bumping Keycloak version to 23.0
- dropping long deprecated constructor
- adding constants that are created over and over again in TS
- making expected log more specific
- fixing OCP args as we can't just join args with empty space, it was alright when only one argument was defined, but each argument needs it's own item in array
- new Keycloak image in Docker doesn't accept args merged with empty space, we need to pass them one by one

Please check the relevant options

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Dependency update
- [x] Refactoring
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update
- [x] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Example scenarios has been updated / added
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by michalvavrik** [comment]
I didn't expect KC bump to require any changes, mmnt.

**Comment by michalvavrik** [comment]
run tests

File: quarkus-test-service-keycloak/src/main/java/io/quarkus/test/services/containers/KeycloakContainerManagedResourceBuilder.java [hunk 1]
```diff
@@ -46,9 +47,9 @@ protected String getExpectedLog() {
     @Override
     public void init(Annotation annotation) {
```
Comment (rsvoboda) [inline quarkus-test-service-keycloak/src/main/java/io/quarkus/test/services/containers/KeycloakContainerManagedResourceBuilder.java:48]: Why is this needed now?

Even for log?

Comment (michalvavrik) [reply quarkus-test-service-keycloak/src/main/java/io/quarkus/test/services/containers/KeycloakContainerManagedResourceBuilder.java:48]: I follow same pattern as is already known from `Container` to keep expectations uniform:

https://github.com/quarkus-qe/quarkus-test-framework/blob/6c458c1060643bf508573eb22da737229b549406/quarkus-test-containers/src/main/java/io/quarkus/test/services/containers/ContainerManagedResourceBuilder.java#L49

Personally I just need this for image, but it is true in theory expected log can be improved with image name as well.

Comment (michalvavrik) [reply quarkus-test-service-keycloak/src/main/java/io/quarkus/test/services/containers/KeycloakContainerManagedResourceBuilder.java:48]: And for image - if you want to have images in `pom.xml`, then you need to resolve them from system properties.

Comment (rsvoboda) [reply quarkus-test-service-keycloak/src/main/java/io/quarkus/test/services/containers/KeycloakContainerManagedResourceBuilder.java:48]: Ok, thanks.

**Comment by jedla97** [comment]
There seems to be Keycloak failure on Openshift with jvm and native as well what I just look. Something wasn't able to start in time. Can you look at it?

**Comment by michalvavrik** [comment]
> There seems to be Keycloak failure on Openshift with jvm and native as well what I just look. Something wasn't able to start in time. Can you look at it?

yes

**Comment by michalvavrik** [comment]
run tests

**Comment by michalvavrik** [comment]
There is a single failure - `OpenShiftGreetingResourceIT` in native. It's impossible for me to run this test as it takes eternity to upload binaries to OCP with my network (30 minutes and counting) but I've checked both native and JVM OCP template of this test - there are no args.

Therefore I suggested the failure cannot be related to changes here. @jedla97 please review the PR.

File: quarkus-test-service-keycloak/src/main/java/io/quarkus/test/bootstrap/KeycloakService.java [hunk 2]
```diff
@@ -11,6 +11,9 @@
 
 public class KeycloakService extends BaseService<KeycloakService> {
 
+    public static final String DEFAULT_REALM_BASE_PATH = "/realms";
+    public static final String DEFAULT_REALM = "test-realm";
+    public static final String DEFAULT_REALM_FILE = "/keycloak-realm.json";
```
Comment (jedla97) [inline quarkus-test-service-keycloak/src/main/java/io/quarkus/test/bootstrap/KeycloakService.java:16]: +1 for adding these in one place.

File: examples/keycloak/src/test/java/io/quarkus/qe/OpenShiftUsingCustomTemplateResourceIT.java [hunk 3]
```diff
@@ -11,12 +15,12 @@
 @OpenShiftScenario
 public class OpenShiftUsingCustomTemplateResourceIT {
 
-    private static final String REALM_DEFAULT = "test-realm";
     private static final String CLIENT_ID_DEFAULT = "test-application-client";
     private static final String CLIENT_SECRET_DEFAULT = "test-application-client-secret";
 
     @Container(image = "quay.io/keycloak/keycloak:22.0.1", expectedLog = "started", port = 8080)
```
Comment (jedla97) [inline examples/keycloak/src/test/java/io/quarkus/qe/OpenShiftUsingCustomTemplateResourceIT.java:21]: Shouldn't here be also newer Keycloak?

**Review by jedla97 — CHANGES_REQUESTED** [review]
Overall it's look good just one comment. And looking through TF I see another usage of Keycloak 22 [here](https://github.com/quarkus-qe/quarkus-test-framework/blob/main/examples/keycloak/src/test/resources/keycloak-custom-template.yaml). If there is need to for these stay on KC22 we can merge this.

File: examples/keycloak/src/test/java/io/quarkus/qe/OpenShiftUsingCustomTemplateResourceIT.java (hunk 3 above)
Comment (michalvavrik) [reply examples/keycloak/src/test/java/io/quarkus/qe/OpenShiftUsingCustomTemplateResourceIT.java:21]: No idea why there is hardcoded image instead of `@KeycloakContainer` used, but I suppose we can do it in a separate PR? I'm happy to change annotation, but I'd like to make release and fix TS.

Also @gtroitsk have some ticket on getting rid of hardcoded images, maybe he can take care of it?

Comment (jedla97) [reply examples/keycloak/src/test/java/io/quarkus/qe/OpenShiftUsingCustomTemplateResourceIT.java:21]: Ok no problem with this. Make sense to leave it the task with removing hardcoded images.

Comment (michalvavrik) [reply examples/keycloak/src/test/java/io/quarkus/qe/OpenShiftUsingCustomTemplateResourceIT.java:21]: Thanks; @gtroitsk please create PR (ideally with others if there are such hardcoded images) and link it with this one.

**Review by jedla97 — APPROVED** [review]
As the older version will be removed in different PR. It's look good and we can merge this.

File: examples/keycloak/src/test/java/io/quarkus/qe/OpenShiftUsingCustomTemplateResourceIT.java (hunk 3 above)
Comment (gtroitsk) [reply examples/keycloak/src/test/java/io/quarkus/qe/OpenShiftUsingCustomTemplateResourceIT.java:21]: I will do it after finish with TS related stuff
